pht_federated/aggregator/services/discovery/column_harmonization.py

```diff
@@ -0,0 +1,222 @@
+from typing import List, Tuple
+
+from thefuzz import fuzz
+
+from pht_federated.aggregator.schemas.dataset_statistics import DatasetStatistics
+
+
+def compare_two_datasets(
+    dataset_statistics: DatasetStatistics,
+    aggregator_statistics: DatasetStatistics,
+    dataset_name: str,
+) -> dict:
+    """
+    Compares two datasets in form of DatasetStatistics objects with respect to their column names
+    :param dataset_statistics: DatasetStatistics of local dataset
+    :param aggregator_statistics: DatasetStatistics of aggregator dataset
+    :param dataset_name: Specified name of local dataset
+    :return: Dictionary which lists the differences between the two datasets
+    """
+
+    input_column_information = dataset_statistics.dict()["column_information"]
+    aggregator_column_information = aggregator_statistics.dict()["column_information"]
+
+    input_column_information = [
+        (x["title"], x["type"]) for x in input_column_information
+    ]
+    aggregator_column_information = [
+        (x["title"], x["type"]) for x in aggregator_column_information
+    ]
+
+    # find intersection
+    (
+        intersection,
+        type_differences,
+        aggregator_column_information,
+        input_column_information,
+    ) = intersection_two_lists(input_column_information, aggregator_column_information)
```

Comment:
Example for object wrapping: instead of returning a tuple of 4 items return for example a ListIntersectionReport object with the corresponding properties.